earth_enterprise/rpms/build.gradle

```diff
@@ -219,7 +212,14 @@ task stageOpenGeeInstall(type: Exec, dependsOn: 'compileOpenGee') {
         "installdir=${stagedInstallDir}")
 }
 
-task openGeePostGisRpm(type: GeeRpm) {
+if (project.hasProperty('buildOpenGee')) {
+    setGeeVersion.dependsOn stageOpenGeeInstall
+}
```

Comment:
This is kind of weird...  But it was what the old code was doing.  Really we should not have a task triggered like this based off of a property.  It makes the gradle command awkward...  A better way might be to have `setGeeVersion` as a `mustRunAfter` to `stageOpenGeeInstall`.  Then if the user wants to have gradle build, stage and package as oppposed to just package (with the assumption that build and stage has already been done) all the user does is run: `gradle stageOpenGeeInstall osPackage` and the right thing would be done.  This is cleaner then having to do `gradle -pbuildOpenGee osPackage`.  But changing the behavior now probably isn't a good idea.

Reply:
I don't think setting the version should depend on staging Open GEE.  Staging has a lot more side effects than generating a version file, if it doesn't exist.  The old code just generated the version file without staging Open GEE.

Reply:
True but there is no task that does anything meaningful with the version that doesn't also need the output of a build and stage available so we might as well (for now) use the build and stage as a means to get the version file.

Reply:
That could be argued.

But, it could also be argued that the only useful tasks in this build file are for generating packages, so it's fine for all of them to depend on building and staging Open GEE.

Reply:
I think using `mustRunAfter` is a good idea, or even detecting if staged install files exist.  I think this option is mostly to make enabling a build from Gradle explicit.  It's also settable from a `.settings` file.